### ISSUE TITLE
K.eval(y_true) inside custom loss function gives Error 

### ISSUE THREAD

**Issue by ahmelshi** [fchollet/keras#7820]
def defined_losss(y_true, y_pred): # y_pred is the generated img
          K.eval(y_true) -->
         return calculation over (y_true & y_pred)

-- > InvalidArgumentError (see above for traceback): Shape [-1,-1,-1,-1] has negative dimensions
	 [[Node: decoder/FinalConv_target = Placeholder[dtype=DT_FLOAT, shape=[?,?,?,?], _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

appreciate any help!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by Dref360** [comment]
You cannot do an eval in a loss function. You can only manipulate tensors.

**Comment by ahmelshi** [comment]
@Dref360 Thank you, now I try to do the following :

def defined_losss(y_true, y_pred): 
y_pred_2 = autoencoder([y_pred, y_true[:,:,0,0]]) --->
return calculation over (y_pred_2 & y_pred)

the autoencoder is a model defined as follows:
autoencoder = Model([encoder.inputs[0], inp_tens], decoder(merged), name=name)

and inp_tens = Input(shape=(3,), name=name)

---> gives compute_output_shape
    assert input_shape[-1]

i mean slicing the y_true still somehow has[-1] values or what is that iam missing ?
